### PR TITLE
apple: Add primitive core audio support

### PR DESCRIPTION
**Pull request by skmp**
This needs to be lock correctly and buffer correctly before making it to master. But it's a place to start!

Sound does work on OS X, on iOS I get silence (but no crashes). iOS uses 1024 frame blocks and that could be one of the issues there

### CODE REVIEW

**Comment by skmp** [comment]
@Holzhaus do you have any ideas on how to fix wrecker? it's really becoming a PITA :<

File: core/oslib/audiobackend_coreaudio.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ */
+
+#include "oslib/audiobackend_coreaudio.h"
+
+#if HOST_OS == OS_DARWIN
+
+//#include <CoreAudio/CoreAudio.h>
+#include <AudioUnit/AudioUnit.h>
+
+AudioUnit audioUnit;
```
Comment (Holzhaus) [inline core/oslib/audiobackend_coreaudio.cpp]: Shouldn't these be static?

Comment (skmp) [inline core/oslib/audiobackend_coreaudio.cpp]: Maybe, but I generally don't use the "static" keyword cus I don't like how it sounds/looks/feels. Linkage control has nothing to do with "staticity". So I add it only when it becomes a problem...

**Comment by mk8itra1n** [comment]
Looks like you got sound to work in reicast. Amazing progress already only days into returning to reicast development. Keep up the good work @skmp

**Pull request merged**